core/commands.py

```diff
@@ -29,22 +29,33 @@ def execute(cls, dataset_slug, tablename, filename, **options):
         data_table = DataTable.new_data_table(table)  # in memory instance, not persisted in the DB
 
         if self.flag_import_data:
+            print(f"Importing data to new table {data_table.db_table_name}")
             Model = self.table.get_model(cache=False, data_table=data_table)
             self.import_data(filename, Model)
         else:
             Model = self.table.get_model(cache=False)
+
+        # Vaccum and concurrent index creation cannot run inside a transaction block
         if self.flag_vacuum:
             self.run_vacuum(Model)
         if self.flag_create_filter_indexes:
             self.create_filter_indexes(Model)
-        if self.flag_fill_choices:
-            self.fill_choices(Model)
+
+        try:
+            with transaction.atomic():
+                if self.flag_fill_choices:
+                    self.fill_choices(Model)
+                if self.flag_import_data:
+                    table.data_table.deactivate(drop_table=self.flag_clean_after)
+                    data_table.activate()
+        except Exception as e:
```

Comment:
Aqui seria legal:
- Colocar pelo menos a última linha da exceção que ocorreu
- Ter certeza de que a `table.data_table` (antiga) está `active`, pois o erro pode ter acontecido entre o `deactivate` e `activate`, ou seja, forçar o estado anterior.

Reply:
Sobre a exceção, beleza vou adicionar isso sim. Sobre o erro, posso ver manualmente aqui, como o `except` tá fora do bloco de transação, então a exceção já invalidou a transação e a desativação não rolou, certo?

Reply:
Péra, relendo aqui eu não entendi o comentário da exceção na real. Eu levanto a exceção após deletar a tabela temporária, então o traceback é apresentado para a pessoa.

Reply:
> Sobre a exceção, beleza vou adicionar isso sim. Sobre o erro, posso ver manualmente aqui, como o `except` tá fora do bloco de transação, então a exceção já invalidou a transação e a desativação não rolou, certo?

Opa, tem razão. Não tinha pensado nisso.